ksqldb-api-client/src/main/java/io/confluent/ksql/api/client/impl/RowImpl.java

```diff
@@ -0,0 +1,149 @@
+/*
+ * Copyright 2020 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.api.client.impl;
+
+import io.confluent.ksql.api.client.Row;
+import io.vertx.core.json.JsonArray;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class RowImpl implements Row {
+
+  private final List<String> columnNames;
+  private final List<String> columnTypes;
+  private final List<Object> values;
+  private final Map<String, Integer> columnNameToIndex;
+
+  @SuppressWarnings("unchecked")
+  public RowImpl(
+      final List<String> columnNames,
+      final List<String> columnTypes,
+      final JsonArray values) {
+    this.columnNames = columnNames;
+    this.columnTypes = columnTypes;
+    this.values = values.getList();
+
+    this.columnNameToIndex = valueToIndexMap(columnNames);
```

Comment:
valueToIndexMap is always the same for each row in a query result, so it seems inefficient to calculate it every time. Instead we could calculate it once and pass the map into the constructor of RowImpl

Reply:
Done. Good point!